scripts/tests/test_zkai_d128_rmsnorm_mlp_fused_gate.py

```
@@ -0,0 +1,136 @@
+
+    def test_atomic_write_preserves_replace_error_when_cleanup_fails(self) -> None:
+        with tempfile.TemporaryDirectory(dir=gate.EVIDENCE_DIR) as tmp:
+            path = Path(tmp) / "replace-error.json"
+            replace_error = OSError("replace failed")
+            with mock.patch.object(gate.os, "replace", side_effect=replace_error):
+                with mock.patch.object(
+                    gate.pathlib.Path,
+                    "unlink",
+                    side_effect=OSError("unlink failed"),
+                ):
+                    with self.assertRaises(OSError) as ctx:
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)